processor/transformprocessor/internal/metrics/functions.go

```diff
@@ -30,10 +31,14 @@ var registry = map[string]interface{}{
 func init() {
 	// Init metrics registry with default functions common to all signals
 	for k, v := range common.Functions[ottldatapoints.TransformContext]() {
-		registry[k] = v
+		datapointRegistry[k] = v
 	}
 }
 
-func Functions() map[string]interface{} {
```

Comment:
This was the root of the issue: the metrics processor was calling this function to initialize the `MetricsParserCollection`, which was then using the registry to parse metric statements.  The issue is that these functions are datapoint context functions, so when used with metric statements things would panic if a metric path was used that isn't in datapoints, such as `name`.